### PR TITLE
Fix Azure http client leak in Azure filesystem

### PR DESCRIPTION
**Pull request by findepi**
Found by https://github.com/trinodb/trino/pull/21913

### CODE REVIEW

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java [hunk 1]
```diff
@@ -58,25 +61,32 @@ public class TestDeltaLakeAdlsConnectorSmokeTest
...
+        String connectionString = format("DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s;EndpointSuffix=core.windows.net", account, accessKey);
+        OkHttpClient okHttpClient = new OkHttpClient.Builder().build();
+        closeAfterClass(() -> {
+            okHttpClient.dispatcher().executorService().shutdownNow();
+            okHttpClient.connectionPool().evictAll();
+        });
+        BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionString)
+                .httpClient(createAzureHttpClient(okHttpClient, new HttpClientOptions()))
+                .buildClient();
+        this.azureContainerClient = blobServiceClient.getBlobContainerClient(container);
```
Comment (charlesjmorgan) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java:88]: Why was the creation of the `azureContainerClient` moved out of the constructor?

Comment (findepi) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java:88]: in the constructor i cannot use `closeAfterClass` yet

File: plugin/trino-delta-lake/pom.xml [hunk 2]
```diff
@@ -261,6 +267,12 @@
             <scope>runtime</scope>
         </dependency>
 
+        <dependency>
+            <groupId>io.trino</groupId>
+            <artifactId>trino-filesystem-azure</artifactId>
+            <scope>runtime</scope>
```
Comment (charlesjmorgan) [inline plugin/trino-delta-lake/pom.xml:273]: Could it be just test scoped? Same for the okhttp dependency above?

File: lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystemFactory.java [hunk 3]
```diff
@@ -65,14 +71,39 @@ public AzureFileSystemFactory(
...
+        int maximumConnectionPoolSize = (poolSize != null && poolSize > 0) ? poolSize : 5;
+
+        return new OkHttpAsyncHttpClientBuilder(okHttpClient)
+                .proxy(clientOptions.getProxyOptions())
+                .configuration(clientOptions.getConfiguration())
+                .connectionTimeout(clientOptions.getConnectTimeout())
+                .writeTimeout(clientOptions.getWriteTimeout())
+                .readTimeout(clientOptions.getReadTimeout())
+                .connectionPool(new ConnectionPool(maximumConnectionPoolSize,
+                        clientOptions.getConnectionIdleTimeout().toMillis(), TimeUnit.MILLISECONDS)).build();
```
Comment (findinpath) [inline lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystemFactory.java:107]: There may be potential for further config options in `AzureFileSystemConfig`
cc @anusudarsan

**Review by findinpath — APPROVED** [review]

File: lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java [hunk 4]
```diff
@@ -116,14 +118,18 @@ private boolean isHierarchicalNamespaceEnabled()
             return fileSystemClient.getDirectoryClient("/").exists();
         }
         catch (RuntimeException e) {
-            throw new IOException("Failed to check whether hierarchical namespaces is enabled for the storage account %s and container %s".formatted(account, containerName));
+            throw new IOException("Failed to check whether hierarchical namespaces is enabled for the storage account %s and container %s" .formatted(account, containerName));
```
[the remark this answers is not in the document]
Comment (findepi) [reply lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java:121]: thanks! for some reason intellij reformat added this for me.
will try to reset the airbase code style.

File: plugin/trino-delta-lake/pom.xml (hunk 2 above)
Comment (findepi) [reply plugin/trino-delta-lake/pom.xml:273]: it is needed by tests, but also needed transitively.
we need to use runtime scope in such case, which sadly disables dep check

File: lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (findepi) [reply lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java]: i reimported airlift codestyle, but intellij still adds this space.
cc @trinodb/maintainers

**Review by losipiuk — APPROVED** [review]

File: lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystemFactory.java (hunk 3 above)
Comment (anusudarsan) [reply lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystemFactory.java:107]: ya we will have test internally with defaults and see if there is a need for it

**Pull request merged**